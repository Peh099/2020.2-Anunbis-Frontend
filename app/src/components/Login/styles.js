import styled from 'styled-components';

export const Message = styled.div`
  color: white;
  font-size: 16px;
  border-radius: 2px;
  text-shadow: rgba(0, 0, 0, 50%) 0px 0px 5px;
  font-weight: bold;
  display: block;
  background: #ffab42;
  padding: 10px;
  box-shadow: 2px 2px rgba(0, 0, 0, 25%);
  max-width: 75%;
  text-align: center;
`;

export const VerifyMailStyle = styled.p`
  button {
    background: transparent;
    border: none;
    font: inherit;
    color: inherit;
    text-shadow: inherit;
    text-decoration: underline;
  }
`;