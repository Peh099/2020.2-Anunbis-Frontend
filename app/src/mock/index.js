import MockAdapter from 'axios-mock-adapter';
import api from '../services/Api';


const mock = new MockAdapter(api);

export default mock