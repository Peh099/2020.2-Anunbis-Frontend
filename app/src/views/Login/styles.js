import styled from 'styled-components';

export const Content = styled.div`
  height: 450px;
  width: 400px;
  display:flex;
  justify-content: space-between;
  align-items: center;
  flex-direction: column;
  cursor: ${(props) => (props.cursor ? props.cursor : '')};
    Form {
    height: 300px;
    width: 300px;

    Button{
      margin-top: 30px;
    }
}
`;

export const Conteiner = styled.div`
      width: 400px;
      height: 15px;
      display:flex;
      align-items: center;
      font-size:14px;
      margin-top: 15px;

      .btnCadastro {
        text-decoration: none;
        color: #212121;
      }

      .btnLogin {
        color: #212121;
        margin: 35px;
      }
  `;

export const Erro = styled.div`
    display: flex;
    flex-direction: row;
    color:black;
    font-size: 14px;
    font-family: Arial, Helvetica, sans-serif;
    background: #f3c2c2;
    border: 1px solid rgba(255, 245, 157, 0.6);
    box-sizing: border-box;
    border-radius: 5px;
    box-shadow: 2px 2px grey;
    Button{
      background: #f3c2c2;
      color: #cf5858;
      border: none;
      box-sizing: border-box;
      border-radius: 3px;
      padding: 5px 5px;
      margin-right: 2px;
    }
`;
