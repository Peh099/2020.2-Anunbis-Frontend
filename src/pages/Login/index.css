.App {
    text-align: center;
  }

.Header{
    background-image: url("../../assets/BG_FGA.png");
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
}

.Login Form{
    width: 100%;
    height: 100%;
    margin-top: 10%;
}

Button{
    margin-top: 30%;
}

Button:hover{
    cursor: pointer;
}

.Erro {
    display: flex;
    color:black;
    font-size: 15px;
    font-family: Arial, Helvetica, sans-serif;
    justify-content: center;
    padding-bottom: 5%;
}

#closeSpan{
    
}