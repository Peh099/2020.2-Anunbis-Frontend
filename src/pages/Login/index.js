import React, {useContext} from "react";
import Button from "../../components/Button";
import Form from "../../components/Form";
import Input from "../../components/Input";
import "./index.css";
import schema from "./validations"

import { useForm } from "react-hook-form";
import { yupResolver } from '@hookform/resolvers/yup';
import AuthContext from '../../contexts/Auth'

export default function Login() {
    const context = useContext(AuthContext);
    const [acessToken, setAcessToken] = React.useState([]);
    const [erroLogin, setErroLogin] = React.useState('');
    const { register, handleSubmit, errors } = useForm({
      resolver: yupResolver(schema),
    });

    function onSubmit(data) {
     const body = {
       email: data.email,
       password: data.password
    }
     console.log(context);
     console.log(data);
     let url = process.env.REACT_APP_API_HOST + "/auth";

     fetch(url, {
       method: 'post',
       headers: { 'Content-Type': 'application/json' },
       body: JSON.stringify(body)
     })
     .then(response => {
        context.Login(response);
     })
    }

    // function createSpanError() {
    //   return (
    //     <div>
    //       {setErroLogin("Email ou Senha Inv√°lidos")}
    //       <Button id="closeSpan" type="button" onClick={setErroLogin} text="X" />
    //     </div>
    //   );
    // }

    return (
      <div className="Login">
        <header className="Header">
          <Form onSubmit={handleSubmit(onSubmit)}>
            <Form.Field><div className="Erro">{erroLogin}</div></Form.Field>
            <Form.Field errorMsg={errors.email?.message}><Input type="text" text="Email Instuticional" name="email" register={register} /> </Form.Field>
            <Form.Field errorMsg={errors.password?.message}><Input type="password" text="Senha" name="password" register={register} /> </Form.Field>
            <Form.Footer>
              <Button text="CONFIRMAR" />
            </Form.Footer>
          </Form>
        </header>
      </div>
    );
}