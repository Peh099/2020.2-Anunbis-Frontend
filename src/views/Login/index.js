import React, { useEffect } from "react";
import styled from 'styled-components';
import { useHistory, Link } from 'react-router-dom';
import { useForm } from "react-hook-form";
import { yupResolver } from '@hookform/resolvers/yup';
import schema from "./validations"
import Button from "../../components/Button";
import Form from "../../components/Form";
import Input from "../../components/Input";
import isAuthenticated, { sendLogin, logOut } from '../../services/Auth'

const Content = styled.div`
  height: 450px;
  width: 400px;
  display:flex;
  justify-content: space-between;
  align-items: center;
  flex-direction: column;
  cursor: ${props => props.cursor ? props.cursor : ""};
    Form {
    height: 300px;
    width: 300px;

    Button{
      margin-top: 30px;
    }
}
`;

const Conteiner = styled.div`
      width: 400px;
      height: 15px;
      display:flex;
      align-items: center;
      font-size:14px;
      margin-top: 15px;

      .btnCadastro {
        text-decoration: none;
        color: #212121;
      }

      .btnLogin {
        color: #212121;
        margin: 35px;
      }
  `;
const Header = ({ children }) => {
  return (
    <Conteiner>
      {children}
    </Conteiner>
  )
}

const Erro = styled.div`
    display: flex;
    flex-direction: row;
    color:black;
    font-size: 14px;
    font-family: Arial, Helvetica, sans-serif;

    background: #f3c2c2;
    border: 1px solid rgba(255, 245, 157, 0.6);
    box-sizing: border-box;
    border-radius: 5px;
    box-shadow: 2px 2px grey;

    Button{
      background: #f3c2c2;
      color: #cf5858;
      border: none;
      box-sizing: border-box;
      border-radius: 3px;
      padding: 5px 5px;
      margin-right: 2px;
    }
`

export default function Login() {
  const history = useHistory();
  const [erroLogin, setErroLogin] = React.useState();
  const [cursor, setCursor] = React.useState();

  const { register, handleSubmit, errors } = useForm({
    resolver: yupResolver(schema),
  });

  useEffect(() => {
    logOut();
  })

  function onSubmit(data) {
    setCursor("wait");
    sendLogin(data.email, data.password, () => {
      if (isAuthenticated())
        history.push("/home");
    }, () => {
      createSpanError();
      setCursor("");
    });
  }

  function createSpanError() {
    setErroLogin(
      <Erro>
        Email ou Senha Inv√°lidos
        <Button type="button" onClick={() => setErroLogin('')} text="X" />
      </Erro>)
  }

  return (
    <Content cursor={cursor}>
      <Header>
        <Link className="btnLogin" to="/user/login">LOGIN</Link>
        <Link className="btnCadastro" to="/user/student">CADASTRO</Link>
      </Header>
      <Form onSubmit={handleSubmit(onSubmit)}>
        <Form.Field><div>{erroLogin}</div></Form.Field>
        <Form.Field errorMsg={errors.email?.message}><Input type="text" text="Email Instuticional" name="email" register={register} /> </Form.Field>
        <Form.Field errorMsg={errors.password?.message}><Input type="password" text="Senha" name="password" register={register} /> </Form.Field>
        <Form.Footer>
          <Button text="CONFIRMAR" backColor="#FFF9C4" cursor={cursor}/>
        </Form.Footer>
      </Form>
    </Content>
  );
}