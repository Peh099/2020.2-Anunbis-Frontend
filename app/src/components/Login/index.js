import React, { useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import schema from './validations';
import Button from '../Button';
import Form from '../Form';
import Input from '../Input';
import { sendLogin, logOut } from '../../services/Auth';
import Users from '../../services/Users';
import { Message, VerifyMailStyle } from './styles';
import api from '../../services/Api';

function Login({ msg }) {
  const history = useHistory();
  const [message, setMessage] = React.useState(msg);
  const [cursor, setCursor] = React.useState();

  const { register, handleSubmit, errors } = useForm({
    resolver: yupResolver(schema),
  });

  useEffect(() => {
    logOut();
  });

  function onSubmit(data) {
    setCursor('wait');
    sendLogin(
      data.email,
      data.password,
      (response) => {
        if (response.status === 200) {
          const home = Users.whoAuthenticated().homePath;
          history.push(home);
        }
        if (response.status === 203) {
          setCursor('pointer');
          setMessage(<VerifyMail email={data.email} />);
        }
      },
      () => {
        setCursor('');
        setMessage('Email ou senha inválido');
      },
    );
  }

  return (
    <Form onSubmit={handleSubmit(onSubmit)}>
      <Form.Field>{message && <Message>{message}</Message>}</Form.Field>
      <Form.Field errorMsg={errors.email?.message}>
        <Input
          type="text"
          text="Email Instuticional"
          name="email"
          register={register}
        />
      </Form.Field>
      <Form.Field errorMsg={errors.password?.message}>
        <Input
          type="password"
          text="Senha"
          name="password"
          register={register}
        />
      </Form.Field>
      <Form.Footer>
        <Button text="CONFIRMAR" backColor="#FFF9C4" cursor={cursor} />
      </Form.Footer>
    </Form>
  );
}

function VerifyMail({ email }) {
  const [message, setMessage] = React.useState()

  function sendVerifyMail() {
    setMessage('Enviando email...');

    api.post('/auth/email', { email })
      .then(response => {
        if (response.status === 200)
          setMessage('E-mail de confirmação enviado.')
      })
  }

  return <VerifyMailStyle>
    {
      message
      || <>Cadastro ainda não confirmado. <button type='submit' onClick={sendVerifyMail}>Clique aqui</button> para reenviar o e-mail de verificação.</>
    }
  </VerifyMailStyle>
}



export default Login;