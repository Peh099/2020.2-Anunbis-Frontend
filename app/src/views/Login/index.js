import React, { useEffect } from 'react';
import { useHistory, Link } from 'react-router-dom';
import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import schema from './validations';
import Button from '../../components/Button';
import Form from '../../components/Form';
import Input from '../../components/Input';
import { sendLogin, logOut, getToken } from '../../services/Auth';
import Users from '../../services/Users';
import { Content, Conteiner, Erro } from './styles';

const Header = ({ children }) => (
  <Conteiner>
    {children}
  </Conteiner>
);

export default function Login() {
  const history = useHistory();
  const [erroLogin, setErroLogin] = React.useState();
  const [cursor, setCursor] = React.useState();

  const { register, handleSubmit, errors } = useForm({
    resolver: yupResolver(schema),
  });

  useEffect(() => {
    logOut();
  });

  function onSubmit(data) {
    setCursor('wait');
    sendLogin(data.email, data.password, () => {
      if (getToken()) {
        const home = Users.PROFESSOR.isAuthenticated() ? 'professor' : 'student';
        history.push(`/${home}`);
      }
    }, () => {
      createSpanError();
      setCursor('');
    });
  }

  function createSpanError() {
    setErroLogin(
      <Erro>
        Email ou Senha Inv√°lidos
        <Button type="button" onClick={() => setErroLogin('')} text="X" />
      </Erro>,
    );
  }

  return (
    <Content cursor={cursor}>
      <Header>
        <Link className="btnLogin" to="/visitant/login">LOGIN</Link>
        <Link className="btnCadastro" to="/visitant/student">CADASTRO</Link>
      </Header>
      <Form onSubmit={handleSubmit(onSubmit)}>
        <Form.Field><div>{erroLogin}</div></Form.Field>
        <Form.Field errorMsg={errors.email?.message}>
          <Input type="text" text="Email Instuticional" name="email" register={register} />
          {' '}
        </Form.Field>
        <Form.Field errorMsg={errors.password?.message}>
          <Input type="password" text="Senha" name="password" register={register} />
          {' '}
        </Form.Field>
        <Form.Footer>
          <Button text="CONFIRMAR" backColor="#FFF9C4" cursor={cursor} />
        </Form.Footer>
      </Form>
    </Content>
  );
}
