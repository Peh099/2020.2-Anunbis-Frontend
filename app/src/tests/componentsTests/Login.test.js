import renderer from 'react-test-renderer';
import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Login from '../../components/Login';
import mock from '../../mock';
import '@testing-library/jest-dom';
import mockLogin from '../../mock/fixtures/login';
import { validStudent } from '../../mock/fixtures/stored_users';

describe('Test Login component', () => {
  describe('Snapshot test', () => {
    it('matches the snapshot', () => {
      const tree = renderer.create(<Login />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  })
  describe('Form submit tests', () => {
    it('To submit email, password fakes must be create div error and close it when click X', async () => { 
      mockLogin('student');

      render(<Login />);
      const inputEmail = screen.getByPlaceholderText('Email Instuticional');
      const inputPassword = screen.getByPlaceholderText('Senha');
      const btnConfirm = screen.getByText('CONFIRMAR');

      userEvent.type(inputEmail, validStudent.email);      
      userEvent.type(inputPassword, validStudent.password);
      userEvent.click(btnConfirm);

      await waitFor(() => {
        expect(screen.queryByText('X')).toBeInTheDocument();
      })
      const btnX = screen.queryByText('X');

      userEvent.click(btnX);

      expect(btnX).not.toBeInTheDocument();
    });
  });
});
