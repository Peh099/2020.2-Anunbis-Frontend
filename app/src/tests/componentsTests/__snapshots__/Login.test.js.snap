// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login component Snapshot test matches the snapshot 1`] = `
<form
  className="sc-hKFxyN okUlG"
  onSubmit={[Function]}
>
  <div
    className="sc-gtsrHT kkYoeF"
  >
    <div
      className="sc-dlnjwi dJBvGq error"
    />
  </div>
  <div
    className="sc-gtsrHT kkYoeF"
  >
    <input
      className="sc-jSFjdj bUyTfK"
      name="email"
      placeholder="Email Instuticional"
      type="text"
    />
    <div
      className="sc-dlnjwi dJBvGq error"
    />
  </div>
  <div
    className="sc-gtsrHT kkYoeF"
  >
    <input
      className="sc-jSFjdj bUyTfK"
      name="password"
      placeholder="Senha"
      type="password"
    />
    <div
      className="sc-dlnjwi dJBvGq error"
    />
  </div>
  <div
    className="sc-eCApnc eIJyFF"
  >
    <button
      className="sc-bdnxRM bIvhHx"
    >
      CONFIRMAR
    </button>
  </div>
</form>
`;
